providers/mandrill/src/lib/mandrill.provider.ts

```diff
@@ -81,4 +95,63 @@ export class MandrillProvider implements IEmailProvider {
       };
     }
   }
+
+  getMessageId(body: any | any[]): string[] {
+    if (Array.isArray(body)) {
+      return body.map((item) => item._id);
+    }
+
+    return [body._id];
+  }
+
+  parseEventBody(
+    body: any | any[],
+    identifier: string
+  ): IEmailEventBody | undefined {
+    if (Array.isArray(body)) {
+      body = body.find((item) => item._id === identifier);
+    }
+
+    if (!body) {
+      return undefined;
+    }
+
+    const status = this.getStatus(body.event);
+
+    if (status === undefined) {
+      return undefined;
+    }
+
+    return {
+      status: status,
+      date: new Date().toISOString(),
+      externalId: body._id,
+      attempts: body.attempt ? parseInt(body.attempt, 10) : 1,
+      response: body.response ? body.response : '',
+      row: body,
+    };
+  }
+
+  private getStatus(event: string): MandrillStatusEnum | undefined {
+    switch (event) {
+      case 'open':
+        return MandrillStatusEnum.OPEN;
+      case 'hard_bounce':
+        return MandrillStatusEnum.HARD_BOUNCE;
+      case 'click':
+        return MandrillStatusEnum.CLICK;
+      case 'send':
+        return MandrillStatusEnum.SEND;
+      case 'spam':
+        return MandrillStatusEnum.SPAM;
+      case 'reject':
+        return MandrillStatusEnum.REJECT;
+      case 'soft_bounce':
+        return MandrillStatusEnum.SOFT_BOUNCE;
+      case 'unsub':
+        return MandrillStatusEnum.UNSUB;
+      case 'deferral':
+        return MandrillStatusEnum.DEFERRAL;
+    }
+  }
```

Comment:
You swapped the wrong thing. The cases are the ones that you need to use the `MandrillStatusEnum` with, and return a status of `EmailStatusEnum` that is the internal of Novu, therefore doing the mapping needed between the 2 systems.

Reply:
But what status should `EmailEventStatusEnum` should return in case of deferral, or